lighthouse-core/runner.js

```diff
@@ -25,6 +25,32 @@ const fs = require('fs');
 const path = require('path');
 const url = require('url');
 
+function runAudit(audit, artifacts) {
+  const status = `Evaluating: ${audit.meta.description}`;
+
+  return Promise.resolve().then(_ => {
+    log.log('status', status);
+
+    // Return an early error if an artifact required for the audit is missing.
+    for (const artifactName of audit.meta.requiredArtifacts) {
+      // TODO: need pass-specific check of networkRecords and traces
```

Comment:
yeah that all makes sense, just wondering if we can nuke the `if (!trace) { throw ...` checks we're adding in here if we just check DEFAULT_PASS where that todo is. Each audit would responsible for checking any non-default on its own for now?